File: apps/party-time-backend/src/main/java/com/partytime/jpa/repository/EventParticipantRepository.java
package com.partytime.jpa.repository;

import com.partytime.jpa.entity.EventParticipant;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;
import java.util.Optional;

public interface EventParticipantRepository extends JpaRepository<EventParticipant, Long> {
    boolean existsByEvent_IdAndAccount_Id(Long eventId, Long accountId);
    Optional<EventParticipant> findByEvent_IdAndAccount_Id(Long eventId, Long accountId);
    Optional<EventParticipant> findByEvent_IdAndAccount_Email(Long eventId, String email);
    List<EventParticipant> findAllByAccount_Email(String email);

}

File: apps/party-time-backend-kotlin/src/main/kotlin/com/partytime/jpa/repository/EventParticipantRepository.kt
package com.partytime.jpa.repository

import com.partytime.jpa.entity.EventParticipant
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.Optional

@Repository
interface EventParticipantRepository: JpaRepository<EventParticipant, Long> {

    fun existsByEvent_IdAndAccount_Id(eventId: Long, accountId: Long): Boolean
    fun findByEvent_IdAndAccount_Id(eventId: Long, accountId: Long): Optional<EventParticipant>
    fun findByEvent_IdAndAccount_Email(eventId: Long, email: String): Optional<EventParticipant>
    fun findAllByAccount_Email(email: String): List<EventParticipant>
}
